docs/cvl/foundry-integration.md

```diff
@@ -0,0 +1,123 @@
+
+# Known Limitations 
+- Foundry's [Invariant testing](https://book.getfoundry.sh/forge/invariant-testing) is not supported, i.e. foundry tests prefixed with 
+`invariant` are not formally verified. Under the hood, the built-in rule `verifyFoundryFuzzTests` is a parametric rule that picks up all methods 
+that start in `test*` and will use these to formally verify them. 
+
+- One of the usual usages of the `setUp()` function is to create new contract instances for testing.
+When setting up the Prover run, the way to handle such storage references to other contracts is to use linking. If, for example,
+ we have the following test code
+
+```solidity
+contract TestContract is Test {
+  MyContract myContract;
+	
+  function setUp() external {
+    myContract = new MyContract();
+  }
+  ...
+}
+```
+
+then add to the `.conf` file:
+
+```json
+  "files": [
+    "...",
+    "path/to/MyContract.sol"
+  ],
+  "link": [
+    "...",
+    "TestContract:myContract=MyContract"
+  ]
+```
+
+This way when the Prover encounters `myContract.foo()` it knows what the implementation of `foo` is and is able to inline it.
+
+- Only a subset of the Foundry cheatcodes are currently implemented.
+The implemented cheatcodes include `vm.assume`, all `assert*` cheatcodes, `vm.expectRevert`, `prank`, `startPrank`, `stopPrank`, 
+`warp`, and `deal`. Some other cheatcodes are irrelevant and are ignored, and many cheatcodes are not yet supported. One could 
+recognize these by checking the **Contracts Call Resolutions** tab of the report - unimplemented cheatcodes will appear there 
+as an unresolved calls (which will usually lead to [havocs](https://docs.certora.com/en/latest/docs/user-guide/glossary.html#term-havoc) 
+and therefore spurious counter examples). In this case, please contact Certora so we can implement the required cheatcode.
+
+- Note that `verifyFoundryFuzzTests` is just one rule that is available to you, the second being `verifyFoundryFuzzTestsNoRevert`. 
+The latter rule ignores reverting paths in the test function (this  is as opposed to the regular definition of foundry tests that 
+will fail if a revert happens in the test function). Foundry tests work by running the test function and failing if the function 
+reverted, and the Prover’s builtin rule `verifyFoundryFuzzTests` will also fail in this case. However, it may happen that there 
+are spurious reverts found by the Prover (e.g. if some call is unresolved, it may assume that the call revert, and the revert propagates up).
+In some cases however, we do not expect the test to ever have an explicit revert occur, but instead want to use the `assert*` 
+cheatcodes to verify some values/state. In this case one could use `verifyFoundryFuzzTestsNoRevert` instead, which completely 
+ignores any path in the code that leads to a revert, but will still verify the `assert*` cheatcodes.
+
+- In Foundry, the `vm.expectRevert` cheatcode could optionally get a specific revert reason and then Foundry will check that 
```

Comment:
Suggestion to rewrite this bullet:

In Foundry, the `vm.expectRevert` cheatcode can optionally take a specific revert reason. Foundry verifies both that the test function reverted and that it reverted with the specified reason. In contrast, the Prover does not analyze the revert reason. As a result, if the test function reverts for a reason other than the expected one, the test will still be marked as successful.